File: core/datacap-spi/src/main/java/io/edurt/datacap/spi/executor/ExecutorModule.java
package io.edurt.datacap.spi.executor;

import com.google.inject.AbstractModule;

public class ExecutorModule
        extends AbstractModule
{
    String name()
    {
        // Intercept the suffix Executor to mark the executor name
        return this.getClass().getSimpleName()
                .replace("Executor", "")
                .replace("Module", "");
    }
}

File: executor/datacap-executor-spi/src/main/kotlin/io/edurt/datacap/executor/ExecutorModule.kt
package io.edurt.datacap.executor

import com.google.inject.AbstractModule

abstract class ExecutorModule : AbstractModule() {
    open fun name(): String {
        return this.javaClass
                .simpleName
                .removeSuffix("Module")
                .removeSuffix("Executor")
    }
}
